### PR TITLE
Make AOV API available at runtime

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Make AOV API available at runtime. Graphic compositor and Automative configurator need it.

Fix fogbugz: https://fogbugz.unity3d.com/f/cases/1265070/
Fix https://fogbugz.unity3d.com/f/cases/1265001/

This PR:
- Add marker for all AOV request operation on GPU
- Add remapping options for Depth Pyramid debug view mode
depth mode
- Fixed AOV export of depth buffer which now correctly export linear depth (case 1265001)
- Add an option to support AOV shader at runtime in HDRP settings (case 1265070) (by not stripping the debug display shader)
- Added error message when allocation from AOV callback allocator fail.

---
### Testing status

QA: 
tested manually that I can build a player for graphic compositor, without the check box it fail to render correctly.
Tested imodifying HDRP_Test 3003_Lightmode to export linear depth. All work correctly.
Automated test for depth buffer export to come in a future PR.

No futher QA required.

Yamato: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fenable-aov-api-at-runtime/.yamato%252Fall-hdrp.yml%2523All_HDRP_fast-trunk/3029313/job/pipeline

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs [hunk 1]
```diff
@@ -101,7 +101,8 @@ internal static RenderPipelineSettings NewDefault() => new RenderPipelineSetting
             decalLayerName7 = "Decal Layer 7",
             msaaSampleCount = MSAASamples.None,
             supportMotionVectors = true,
-            supportRuntimeDebugDisplay = true,
+            supportRuntimeDebugDisplay = false,
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs:104]: Is it intended to change the default here for debug display?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs:104]: yes. We shouldn't compile runtimedebugdisplay by default, big overred for compil time

File: com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs [hunk 2]
```diff
@@ -169,6 +169,10 @@ public class DebugData
             public float debugOverlayRatio = 0.33f;
             /// <summary>Current full screen debug mode.</summary>
             public FullScreenDebugMode fullScreenDebugMode = FullScreenDebugMode.None;
+            /// <summary>Enable range remapping.</summary>
+            public bool enableDebugRemap = false;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs:173]: That's only for depth right? Shouldn't we call this with something more explicit then? enableDebugDepthRemap or something like that?

File: com.unity.render-pipelines.high-definition/Runtime/Debug/DebugFullScreen.shader [hunk 3]
```diff
@@ -28,6 +28,7 @@ Shader "Hidden/HDRP/DebugFullScreen"
 
             CBUFFER_START (UnityDebug)
             float _FullScreenDebugMode;
+            float4 _FullScreenDebugRemap;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/DebugFullScreen.shader:31]: Same thing as above

File: com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Options.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ To expose these properties, select a Composition Layer in the **Render Schedule*
...
 | **Output Renderer**     | Specifies a Renderer to direct the output of this layer to. The compositor overrides and automatically updates the **_BaseColorMap** Texture from the Material attached to this Renderer. This is useful when the selected Renderer should be visible by a Camera on another layer. |
-| **AOVs**                | Specifies the type of output variable in this layer. Aside from Color, you can also output variables like **Albedo**, **Normal**, or **Smoothness**. This option affects all Cameras stacked in this layer. |
+| **AOVs**                | Specifies the type of output variable in this layer. Aside from Color, you can also output variables like **Albedo**, **Normal**, or **Smoothness**. This option affects all Cameras stacked in this layer. For this functionnality to work in the player, it is required to enable **Runtime AOV API** in the [HDRP Asset](HDRP-Asset.md)  |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Options.md:26]: To make this functionality work in the Unity Player, enable **Runtime AOV API** in the [HDRP Asset](HDRP-Asset.md).

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Asset.md [hunk 5]
```diff
@@ -40,7 +40,8 @@ When you create an HDRP Asset, open it in the Inspector to edit its properties.
...
 | **Motion Vectors**                      | Enable the checkbox to make HDRP support motion vectors. HDRP uses motion vectors for effects like screen space reflection (SSR) and motion blur. When disabled, motion blur has no effect and HDRP calculates SSR with lower quality. |
-| **Runtime Debug Display**               | Enable the checkbox to make HDRP display Material and Lighting properties at runtime to help debugging.Disable this checkbox to reduce build time and Shader memory. This disables the following debug modes: All Material debug modes except GBuffer debug. The Lux meter, diffuse lighting only, and specular lighting only debug modes. The overriding option for overriding albedo. |
+| **Runtime Debug Display**               | Enable the checkbox to make HDRP being able to use debug mode from RenderPipeline windows at runtime. Disable this checkbox to reduce build time and Shader memory. This disables the following debug modes: All material properties debug modes except GBuffer debug. The various properties overrides options. All the lighting debug modes.  |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/HDRP-Asset.md:43]: Enable the checkbox to make HDRP able to use debug modes from the Render Pipeline Debug window at runtime. Disable the checkbox to reduce build time and shader memory. This disables the following debug modes: All material property debug modes except GBuffer debug, the various property override options, and all the lighting debug modes.

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Asset.md [hunk 6]
```diff
@@ -40,7 +40,8 @@ When you create an HDRP Asset, open it in the Inspector to edit its properties.
...
+| **Runtime AOV API**                     | Enable the checkbox to make HDRP being able to use AOV API (Rendering of Material properties an lighting mode) at runtime. Disable this checkbox to reduce build time and Shader memory. This disables the following AOV modes: All material properties and Lighting modes. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/HDRP-Asset.md:44]: Enable the checkbox to make HDRP able to use the AOV API (rendering of material properties and lighting modes) at runtime. Disable this checkbox to reduce build time and shader memory. This disables the following AOV modes: All material properties and lighting modes.

**Comment by sebastienlagarde** [comment]
adressed PR feedback, merging.

**Pull request merged**